datafusion/sqllogictest/test_files/functions.slt

```diff
@@ -193,11 +193,10 @@ SELECT substr('alphabet', 3, CAST(NULL AS int))
 ----
 NULL
 
-statement error The first argument of the substr function can only be a string, but got Int64
-SELECT substr(1, 3)
-
-statement error The first argument of the substr function can only be a string, but got Int64
-SELECT substr(1, 3, 4)
+query T
+select substr(123456, 3);
+----
+3456
```

Comment:
`string/string_literal.slt` is used to collect the string-related test. It's better to move it there.